src/schemas/message.ts

```diff
@@ -76,29 +76,12 @@ export interface MessagePayloadBase {
   text?         : string,
   timestamp     : number,        // Unix Timestamp(in seconds)
   type          : MessageType,
-}
-
-/** @hidden */
-export interface MessagePayloadRoom {
   fromId?        : string,
-  // mentionIdList? : string[],   // Mentioned Contacts' Ids
-  roomId         : string,
+  mentionIdList? : string[],   // Mentioned Contacts' Ids
+  roomId?        : string,
   toId?          : string,     // if to is not set, then room must be set
 }
 
-/** @hidden */
-export interface MessagePayloadTo {
```

Comment:
Please do not simply merge those interfaces. They are separated by design.

We need to make sure that: **either `fromId` or `roomId` must exist**

The original design is to protect those data schema; please do not remove this protection.

Reply:
Okay, if that's the case, we should have no `roomId` in the `MessagePayloadTo`, and no `fromId` or `toId` in the `MessagePayloadRoom`, right?